### PR TITLE
feat: dashboards backend

### PR DESCRIPTION
**Pull request by TRG-T**

### CODE REVIEW

File: packages/backend/src/infrastructure/express/router/dashboard.routes.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  constructor(app: express.Application) {
+    super(app, 'UsersRoutes')
+  }
+
+  // prettier-ignore
+  configureRoutes(): express.Application {
+    this.app
+      .route(`${V1}/`)
+      .get(
+          (req: express.Request, res: express.Response) => res.send('Hello World!')
```
Comment (mcwiekala) [inline packages/backend/src/infrastructure/express/router/dashboard.routes.ts:17]: this hello world can be deleted ;)

File: packages/backend/src/infrastructure/express/router/dashboard.routes.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+       );
+
+    this.app
+      .route(`${V1}/dashboards`)
+      .get(
+        dashboardController.getDashboards
+      )
+
+    this.app
+      .route(`${V1}/create-dashboard`)
```
Comment (mcwiekala) [inline packages/backend/src/infrastructure/express/router/dashboard.routes.ts:27]: change to: `${V1}/dashboards`

File: packages/backend/src/infrastructure/express/router/dashboard.routes.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      .post(
+        dashboardController.createDashboard
+      )
+
+    this.app
+      .route(`${V1}/dashboard/:id`)
```
Comment (mcwiekala) [inline packages/backend/src/infrastructure/express/router/dashboard.routes.ts:33]: change to: `${V1}/dashboards/:id`

File: packages/backend/src/modules/dashboard/dashboard.controller.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import CreateDashboardCommand from 'packages/shared/src/api/dto/CreateDashboardCommand.dto'
+import express from 'express'
+import dashboardService from './dashboard.service'
+
+class DashboardController {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.controller.ts:5]: use constructor and `private readonly` fields for service and mapper

File: packages/backend/src/modules/dashboard/dashboard.controller.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  // eslint-disable-next-line class-methods-use-this
+  async createDashboard(req: express.Request, res: express.Response) {
+    console.log(`Received new dashboard with title: ${req.body.title}`)
+    const createDashboardCommand: CreateDashboardCommand = req.body
+    const Dashboard = await dashboardService.createDashboard(createDashboardCommand)
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.controller.ts:10]: use TS types, here Dashboard and then map to DashboardDTO

File: packages/backend/src/modules/dashboard/dashboard.controller.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    return res.status(200).send(Dashboard.id)
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.controller.ts:11]: use 201 status for create, and return created daschboard with id generated in mongo

File: packages/backend/src/modules/dashboard/dashboard.controller.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  async getDashboards(req: express.Request, res: express.Response) {
+    const Dashboard = await dashboardService.getDashboards()
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.controller.ts:16]: use TS types, here Dashboard and then map to DashboardDTO

File: packages/backend/src/modules/dashboard/dashboard.controller.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    return res.status(200).json(Dashboard)
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  async getDashboard(req: express.Request, res: express.Response) {
+    const Dashboard = await dashboardService.getDashboard(req.params.id)
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.controller.ts:22]: use TS types, here Dashboard and then map to DashboardDTO

File: packages/backend/src/modules/dashboard/dashboard.model.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
+import mongoose from 'mongoose'
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.model.ts:1]: why this file is looking like new?

File: packages/backend/src/modules/dashboard/dashboard.model.ts [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  users: [
+    {
+      _id: ObjectId,
+      username: String,
+      avatarId: Number
+    }
+  ],
+  columns: [columnSchema]
+})
+const Dashboard = mongoose.model('Board', dashboardSchema)
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.model.ts:40]: Dashboard not Board

Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.model.ts:40]: Dashboard not Board

File: packages/backend/src/modules/dashboard/dashboard.repository.ts [hunk 11]
```diff
@@ -1,16 +1,54 @@
...
-import DashboardModel from './Dashboard'
+import CreateDashboardCommand from 'packages/shared/src/api/dto/CreateDashboardCommand.dto'
+import CreateTaskCommand from 'packages/shared/src/api/dto/CreateTaskCommand.dto'
+import mongoose from 'mongoose'
+import { Dashboard, DashboardColumn } from './dashboard.model'
 import { Task } from '../task/task'
 
 export class DashboardRepository {
-  private readonly _dashboardModel = DashboardModel
+  private readonly dashboardModel = Dashboard
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.repository.ts:8]: initialize in constructor

File: packages/backend/src/modules/dashboard/dashboard.repository.ts [hunk 12]
```diff
@@ -1,16 +1,54 @@
...
   constructor(boardModel: any) {
-    this._dashboardModel = boardModel
+    this.dashboardModel = boardModel
   }
 
-  async addNewTaskToDashboard(createTaskCommand: CreateTaskCommandDTO, savedTask: Task) {
-    const dashboard = await this._dashboardModel.findById(createTaskCommand.idDashboard)
+  async createDashboard(createDashboardCommand: CreateDashboardCommand) {
+    console.log('Creating new dashboard in DB')
+    const dashboardColumn = DashboardColumn.create({
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.repository.ts:16]: why do you create column here?

File: packages/backend/src/modules/dashboard/dashboard.repository.ts [hunk 13]
```diff
@@ -1,16 +1,54 @@
...
+      },
+      users: [],
+      columns: [dashboardColumn, dashboardColumn]
+    })
+
+    await dashboard.save()
+    return dashboard
+  }
+
+  async getDashboards() {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.repository.ts:38]: add return type to method

File: packages/backend/src/modules/dashboard/dashboard.repository.ts [hunk 14]
```diff
@@ -1,16 +1,54 @@
...
+  private readonly dashboardModel = Dashboard
 
   constructor(boardModel: any) {
-    this._dashboardModel = boardModel
+    this.dashboardModel = boardModel
   }
 
-  async addNewTaskToDashboard(createTaskCommand: CreateTaskCommandDTO, savedTask: Task) {
-    const dashboard = await this._dashboardModel.findById(createTaskCommand.idDashboard)
+  async createDashboard(createDashboardCommand: CreateDashboardCommand) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.repository.ts:14]: add return type to method

File: packages/backend/src/modules/dashboard/dashboard.repository.ts [hunk 15]
```diff
@@ -1,16 +1,54 @@
...
+    return dashboard
+  }
+
+  async getDashboards() {
+    const boards = await this.dashboardModel.find()
+    console.log(boards)
+    return boards
+  }
+
+  async getDashboard(_id: string) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.repository.ts:44]: add return type to method

File: packages/backend/src/modules/dashboard/dashboard.repository.ts [hunk 16]
```diff
@@ -1,16 +1,54 @@
...
+    const dashboard = await this.dashboardModel.findById(_id)
+    console.log(dashboard)
+    return dashboard
+  }
+
+  async addNewTaskToDashboard(createTaskCommand: CreateTaskCommand, savedTask: Task) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.repository.ts:50]: add return type to method

File: packages/backend/src/modules/dashboard/dashboard.service.ts [hunk 17]
```diff
@@ -0,0 +1,22 @@
+import CreateDashboardCommand from 'packages/shared/src/api/dto/CreateDashboardCommand.dto'
+import dashboardRepository from './dashboard.repository'
+
+class DashboardService {
+  // eslint-disable-next-line class-methods-use-this
+  createDashboard(createDashboardCommand: CreateDashboardCommand) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.service.ts:6]: add return type to method

Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.service.ts:6]: add return type to method

File: packages/backend/src/modules/dashboard/dashboard.service.ts [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+    console.log('Handling new dashboard')
+    return dashboardRepository.createDashboard(createDashboardCommand)
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getDashboards() {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.service.ts:12]: add return type to method

File: packages/backend/src/modules/dashboard/dashboard.service.ts [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+    return dashboardRepository.getDashboards()
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getDashboard(_id: string) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.service.ts:17]: add return type to method

File: packages/backend/src/modules/dashboard/dashboard.service.ts [hunk 20]
```diff
@@ -0,0 +1,22 @@
+import CreateDashboardCommand from 'packages/shared/src/api/dto/CreateDashboardCommand.dto'
+import dashboardRepository from './dashboard.repository'
+
+class DashboardService {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.service.ts:4]: use constructor and private readonly fields for service and mapper

File: packages/backend/src/modules/task/task.mapper.ts [hunk 21]
```diff
@@ -1,4 +1,4 @@
-import { TaskDTO } from 'shared'
+import TaskDTO from 'packages/shared/src/api/dto/task.dto'
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.mapper.ts:1]: you should be able to import it just from 'shared'

File: packages/backend/src/modules/task/task.repository.ts [hunk 22]
```diff
@@ -1,4 +1,5 @@
-import { CreateTaskCommandDTO, TaskDTO } from 'shared'
+import { CreateTaskCommandDTO } from 'packages/shared/src/api/dto/CreateTaskCommand.dto'
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.repository.ts:1]: you should be able to import it just from 'shared'

File: packages/backend/src/modules/task/task.service.ts [hunk 23]
```diff
@@ -1,4 +1,4 @@
-import { CreateTaskCommandDTO, TaskDTO } from 'shared'
+import { CreateTaskCommandDTO } from 'packages/shared/src/api/dto/CreateTaskCommand.dto'
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.service.ts:1]: you should be able to import it just from 'shared'

File: packages/frontend/src/infrastructure/HttpService/HttpService.ts [hunk 24]
```diff
@@ -1,4 +1,4 @@
-import { TASK } from 'shared'
+import { TASK } from 'packages/shared/src/paths/task'
```
Comment (mcwiekala) [inline packages/frontend/src/infrastructure/HttpService/HttpService.ts:1]: you should be able to import it just from 'shared'

File: packages/shared/src/api/dto/CreateDashboardCommand.dto.ts [hunk 25]
```diff
@@ -0,0 +1,13 @@
+import { Status } from 'packages/backend/src/modules/dashboard/Status'
```
Comment (mcwiekala) [inline packages/shared/src/api/dto/CreateDashboardCommand.dto.ts:1]: move status to shared since it will be used on frontend and backend

Comment (mcwiekala) [reply packages/shared/src/api/dto/CreateDashboardCommand.dto.ts:1]: and rename it to sth more meaningful maybe 'DashboardVisibility'?

File: packages/shared/src/api/dto/CreateDashboardCommand.dto.ts [hunk 26]
```diff
@@ -0,0 +1,13 @@
...
+
+export interface CreateTaskCommand {
+  title: string
+  description: string
+  coverImageUrl: string
+  status: {
+    type: string
```
Comment (mcwiekala) [inline packages/shared/src/api/dto/CreateDashboardCommand.dto.ts:8]: what is this vaule for?

File: packages/backend/src/infrastructure/express/router/dashboard.routes.ts [hunk 27]
```diff
@@ -0,0 +1,40 @@
...
+    this.app
+      .route(`${V1}/`)
+      .get(
+          (req: express.Request, res: express.Response) => res.send('Hello World!')
+       );
+
+    this.app
+      .route(`${V1}/dashboards`)
+      .get(
+        dashboardController.getDashboards
```
Comment (mcwiekala) [inline packages/backend/src/infrastructure/express/router/dashboard.routes.ts:23]: use here controller object instead of static method declaration

File: packages/backend/src/modules/dashboard/dashboard.model.ts (hunk 9 above)
Comment (TRG-T) [reply packages/backend/src/modules/dashboard/dashboard.model.ts:1]: Before there was `Dashboard.ts`, now there is `dashboard.model.ts`

File: packages/backend/package.json [hunk 28]
```diff
@@ -11,7 +11,7 @@
     "express": "^4.17.2",
     "express-mongo-sanitize": "^2.2.0",
     "mongoose": "^6.2.1",
-    "shared": "*"
+    "shared": "file:./../shared"
```
Comment (mcwiekala) [inline packages/backend/package.json:14]: why?

File: packages/backend/src/modules/dashboard/dashboard.service.ts [hunk 29]
```diff
@@ -0,0 +1,30 @@
...
+import { Dashboard } from './Dashboard'
+
+export class DashboardService {
+  private readonly dashboardRepository: DashboardRepository
+
+  constructor(dashboardRepository: DashboardRepository) {
+    console.log('DashboardService constructor')
+    this.dashboardRepository = dashboardRepository
+  }
+  // eslint-disable-next-line class-methods-use-this
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.service.ts:12]: this rule shoud be ignored right now

File: packages/backend/tsconfig.json [hunk 30]
```diff
@@ -12,7 +12,8 @@
     "moduleResolution": "node",
     "resolveJsonModule": true,
     "isolatedModules": true,
-    "noEmit": true
+    "noEmit": true,
+    "types": ["node"]
   },
-  "include": ["src"]
+  "include": ["src", "../shared/Status.ts"]
```
Comment (mcwiekala) [inline packages/backend/tsconfig.json:18]: lerna should do the trick with importing shared files

File: packages/shared/src/api/DashboardVisibility.ts [hunk 31]
```diff
@@ -0,0 +1,6 @@
+enum DashboardVisibility {
+  PUBLIC = 'PUBLIC',
```
Comment (mcwiekala) [inline packages/shared/src/api/DashboardVisibility.ts:2]: there is no need to to set the vaule exactly like the name. PUBLIC, PRIVATE should be enough

File: packages/backend/src/modules/dashboard/dashboard.model.ts [hunk 32]
```diff
@@ -0,0 +1,39 @@
...
+  ]
+})
+
+const dashboardSchema = new mongoose.Schema({
+  _id: ObjectId,
+  title: String,
+  description: String,
+  imageCoverUrl: String,
+  createdAt: Date,
+  status: DashboardVisibility,
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.model.ts:27]: is it working? I'vs got:
trello-backend: TypeError: Invalid schema configuration: `PUBLIC` is not a valid type at path `status.PUBLIC`. See https://bit.ly/mongoose-schematypes for a list of valid schema types.

File: packages/backend/src/infrastructure/express/router/dashboard.routes.ts [hunk 33]
```diff
@@ -0,0 +1,25 @@
+import express from 'express'
+import { CommonRoutesConfig } from './common.routes.config'
+import dashboardController from '../../../modules/dashboard/dashboard.controller'
+
+const V1 = '/v1'
```
Comment (kacper-cyra) [inline packages/backend/src/infrastructure/express/router/dashboard.routes.ts:5]: I wrote this on some other review, so please add comment:
`// TODO change to value from config`

File: packages/backend/src/modules/dashboard/dashboard.controller.ts [hunk 34]
```diff
@@ -0,0 +1,42 @@
...
+
+class DashboardController {
+  private readonly dashboardMapper: DashboardMapper
+  private readonly dashboardService: DashboardService
+
+  constructor(dashboardService: DashboardService, dashboardMapper: DashboardMapper) {
+    this.dashboardMapper = dashboardMapper
+    this.dashboardService = dashboardService
+  }
+  // eslint-disable-next-line class-methods-use-this
```
Comment (kacper-cyra) [inline packages/backend/src/modules/dashboard/dashboard.controller.ts:16]: Disable for whole file or make a static class

File: packages/backend/src/modules/dashboard/dashboard.repository.ts [hunk 35]
```diff
@@ -1,15 +1,49 @@
...
 import { Task } from '../task/task'
+import { Dashboard } from './Dashboard'
 
 export class DashboardRepository {
-  private readonly _dashboardModel = DashboardModel
+  private readonly _dashboardModel
 
-  constructor(boardModel: any) {
-    this._dashboardModel = boardModel
+  constructor(dashboardModel: any) {
```
Comment (kacper-cyra) [inline packages/backend/src/modules/dashboard/dashboard.repository.ts:10]: You can make it, same as [here ](https://github.com/kacper-cyra/camp-trello-clone/pull/113/files#diff-44f96226e0025aa1eb9c81007908cc30c0e8c76ea08c893de43963b5e566f9a7)to avoid any

File: packages/backend/src/modules/dashboard/dashboard.repository.ts [hunk 36]
```diff
@@ -1,15 +1,49 @@
...
+
+    await dashboard.save()
+    return dashboard
+  }
+
+  async getDashboards(): Promise<Dashboard[]> {
+    const dashboards: Dashboard[] = await this._dashboardModel.find()
+    console.log(dashboards)
+    return dashboards
+  }
```
Comment (kacper-cyra) [inline packages/backend/src/modules/dashboard/dashboard.repository.ts:35]: If it does not work, it would be safer to comment it out for now

File: packages/backend/src/modules/dashboard/dashboard.repository.ts [hunk 37]
```diff
@@ -1,15 +1,49 @@
...
+    console.log(dashboard)
+    return dashboard
+  }
+
+  async addNewTaskToDashboard(
+    createTaskCommand: CreateTaskCommandDTO,
+    savedTask: Task
+  ): Promise<Task> {
     const dashboard = await this._dashboardModel.findById(createTaskCommand.idDashboard)
     const idCol = { idColumn: createTaskCommand.idColumn }
```
Comment (kacper-cyra) [inline packages/backend/src/modules/dashboard/dashboard.repository.ts:48]: I know, it's not yours, but if we don't use it as an object, than there is no need to create one. And there is a rule - leave the code in better state than it was 😄

File: packages/backend/tsconfig.json [hunk 38]
```diff
@@ -1,7 +1,7 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "target": "es5",
+    "target": "es2017",
```
Comment (kacper-cyra) [inline packages/backend/tsconfig.json:4]: What was the problem?

File: packages/frontend/src/infrastructure/HttpService/HttpService.ts (hunk 24 above)
Comment (kacper-cyra) [reply packages/frontend/src/infrastructure/HttpService/HttpService.ts:1]: I agree, if there is an problem, we can try to work it out

**Review by kacper-cyra — CHANGES_REQUESTED** [review]

File: postman/camp-trello-clone.postman_collection.json [hunk 39]
```diff
@@ -1,102 +1,185 @@
 {
-  "info": {
```
Comment (mcwiekala) [inline postman/camp-trello-clone.postman_collection.json:2]: add separate folder just for dashboards. Don't mix tasks with daschboards

File: packages/backend/src/index.ts [hunk 40]
```diff
@@ -1,14 +1,14 @@
...
 import http from 'http'
+import { Dashboard } from './modules/dashboard/dashboard.model'
 import Task from './modules/task/task.model'
-import Dashboard from './modules/dashboard/Dashboard'
 import app from './app'
 import { connectToDatabase } from './infrastructure/mongoose'
 import User from './modules/user/User'
 import Attachment from './modules/attachment/Attachment'
 import { CommonRoutesConfig } from './infrastructure/express/router/common.routes.config'
-import { TaskRoutes } from './infrastructure/express/router/task.routes'
```
Comment (mcwiekala) [inline packages/backend/src/index.ts:11]: why did you remove TaskRoute?

File: packages/backend/src/index.ts [hunk 41]
```diff
@@ -49,7 +49,7 @@ const startServer = async () => {
   const server: http.Server = http.createServer(app)
   const routes: Array<CommonRoutesConfig> = []
 
-  routes.push(new TaskRoutes(app))
```
Comment (mcwiekala) [inline packages/backend/src/index.ts:52]: why did you remove TaskRoute?

File: packages/backend/src/modules/dashboard/Dashboard.ts [hunk 42]
```diff
@@ -1,40 +1,23 @@
...
-  ],
-  columns: [columnSchema]
-})
-const Dashboard = mongoose.model('Dashboard', dashboardSchema)
-export default Dashboard
+  public get title() {
+    return this._title
+  }
+
+  // TODO:
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/Dashboard.ts:18]: what about this TODO? Remove or uncomment

File: packages/backend/src/modules/dashboard/Dashboard.ts [hunk 43]
```diff
@@ -1,40 +1,23 @@
-import mongoose from 'mongoose'
-import { Status } from './Status'
+export class Dashboard {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/Dashboard.ts:1]: follow filename convention: lowercase daschboard.ts

File: packages/backend/src/modules/dashboard/dashboard.controller.ts [hunk 44]
```diff
@@ -0,0 +1,42 @@
...
+  async getDashboards(req: express.Request, res: express.Response) {
+    const dashboards: Dashboard[] = await dashboardService.getDashboards()
+    const dashboardsDTO: DashboardDTO[] = dashboards.map((dashboard: Dashboard) =>
+      this.dashboardMapper.mapToDto(dashboard)
+    )
+    return res.status(200).json(dashboardsDTO)
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  async getDashboard(req: express.Request, res: express.Response) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.controller.ts:35]: maybe rename to: getDashboardById?

Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.controller.ts:35]: maybe rename to: getDashboardById?

File: packages/backend/src/modules/task/task.controller.ts [hunk 45]
```diff
@@ -1,4 +1,5 @@
-import { CreateTaskCommandDTO, TaskDTO } from 'shared'
+import { CreateTaskCommandDTO } from 'packages/shared/src/api/dto/CreateTaskCommand.dto'
+import TaskDTO from 'packages/shared/src/api/dto/task.dto'
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.controller.ts:2]: use just shared

File: packages/shared/src/api/dto/dashboard.dto.ts [hunk 46]
```diff
@@ -0,0 +1,9 @@
+interface DashboardDTO {
+  _id?: string
```
Comment (mcwiekala) [inline packages/shared/src/api/dto/dashboard.dto.ts:2]: rename jest to id. Its just a DTO optional type

File: packages/backend/src/modules/dashboard/dashboard.repository.ts (hunk 35 above)
Comment (TRG-T) [reply packages/backend/src/modules/dashboard/dashboard.repository.ts:10]: The link leads to a file in which this piece of code looks the same

File: packages/shared/src/api/dto/CreateDashboardCommand.dto.ts [hunk 47]
```diff
@@ -0,0 +1,10 @@
+import DashboardVisibility from '../DashboardVisibility'
+
+export interface CreateDashboardCommand {
```
Comment (mcwiekala) [inline packages/shared/src/api/dto/CreateDashboardCommand.dto.ts:3]: move to: packages/shared/src/api/command and call it createDashboard.command.ts

File: packages/backend/src/modules/dashboard/dashboard.controller.ts [hunk 48]
```diff
@@ -0,0 +1,39 @@
...
+    const dashboards: Dashboard[] = await dashboardService.getDashboards()
+    const dashboardsDTO: DashboardDTO[] = dashboards.map((dashboard: Dashboard) =>
+      this.dashboardMapper.mapToDto(dashboard)
+    )
+    return res.status(200).json(dashboardsDTO)
+  }
+
+  async getDashboardById(req: express.Request, res: express.Response) {
+    const dashboard: Dashboard = await dashboardService.getDashboard(req.params.id)
+    const dashboardDTO: DashboardDTO = this.dashboardMapper.mapToDto(dashboard)
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.controller.ts:34]: what if there is no dashboard with given id, or something else happen? The whole app will crash...

add try catch and return 400 in case of any error

File: packages/backend/src/modules/dashboard/dashboard.model.ts [hunk 49]
```diff
@@ -0,0 +1,44 @@
+import mongoose from 'mongoose'
+import { DashboardVisibility } from 'shared'
+import { Dashboard as DashboardB } from './dashboard'
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.model.ts:3]: what? why?

File: packages/backend/src/index.ts [hunk 50]
```diff
@@ -1,14 +1,17 @@
...
-import Dashboard from './modules/dashboard/Dashboard'
 import app from './app'
 import { connectToDatabase } from './infrastructure/mongoose'
 import User from './modules/user/User'
 import Attachment from './modules/attachment/Attachment'
 import { CommonRoutesConfig } from './infrastructure/express/router/common.routes.config'
 import { TaskRoutes } from './infrastructure/express/router/task.routes'
+import { DashboardRoutes } from './infrastructure/express/router/dashboard.routes'
+
+dotenv.config()
```
Comment (mcwiekala) [inline packages/backend/src/index.ts:14]: compiltaion error. How you want to use it without import?

File: packages/backend/src/modules/dashboard/dashboard.repository.ts [hunk 51]
```diff
@@ -1,19 +1,58 @@
...
   }
 
-  async addNewTaskToDashboard(createTaskCommand: CreateTaskCommandDTO, savedTask: Task) {
+  async createDashboard(createDashboardCommand: CreateDashboardCommand): Promise<Dashboard> {
+    console.log('Creating new dashboard in DB')
+    const dashboard = await this._dashboardModel.create({
+      _id: new mongoose.Types.ObjectId(),
+      title: createDashboardCommand.title,
+      description: createDashboardCommand.description,
+      imageCoverUrl: createDashboardCommand.imageCoverUrl,
```
Comment (mcwiekala) [inline packages/backend/src/modules/dashboard/dashboard.repository.ts:20]: this Command doesnt has this filed:
TS2339: Property 'imageCoverUrl' does not exist on type 'CreateDashboardCommand'.

Comment (TRG-T) [reply packages/backend/src/modules/dashboard/dashboard.repository.ts:20]: I have no such problem. Maybe using `yarn libbuild` in the shared package will solve this?

Comment (mcwiekala) [reply packages/backend/src/modules/dashboard/dashboard.repository.ts:20]:
```
{
    _id: ObjectId('622e5dd9c3cba876b8800f93'),
    title: 'Tablica 1',
    description: 'Opis tablicy pierwszej',
    createdAt: 'Fri May 06 2022 20:16:40 GMT+0000 (UTC)',
    status: 'PUBLIC',
    users: [
        {
            _id: ObjectId('622e5dffa63863cfc5f812f5'),
            username: 'Marek Kowalski',
            avatarId: 345
        }
    ],
    columns: [
        {
            _id: ObjectId('622e5e084eb135ee631dcd8e'),
            title: 'Kolumna Pierwsza',
            tasks: [
                {
                    _id: ObjectId('622e5e109a5d1fb63a360fb0'),
                    title: 'Tytuł 1',
                    imageCoverId: 'photo-1646388023469-18941b7e3b6f',
                    comments: [
                        {
                            _id: ObjectId('622e5e179c945199e651891a')
                        }
                    ],
                    attachments: [
                        {
                            _id: ObjectId('622e5e1c15be96a29a180abe')
                        }
                    ],
                    users: [
                        {
                            _id: ObjectId('622e5e2094298d670e6129ee')
                        }
                    ]
                }
            ]
        }
    ]
}
```

**Review by mcwiekala — DISMISSED** [review]

**Review by mcwiekala — APPROVED** [review]

**Review by mcwiekala — APPROVED** [review]
